apps/qubit/modules/informationobject/actions/updatePublicationStatusAction.class.php

```diff
@@ -35,19 +35,60 @@ public function execute($request)
             if ($this->form->isValid()) {
                 // Update resource synchronously
                 $publicationStatusId = $this->form->getValue('publicationStatus');
-                $this->resource->setPublicationStatus($publicationStatusId);
-                $this->resource->save();
 
-                // Update descendants using job scheduler
+                $sql = 'UPDATE status
+                    JOIN information_object io ON status.object_id = io.id
+                    SET status.status_id = :publicationStatus
+                    WHERE status.type_id = :publicationStatusType
+                    AND io.id = :id';
+
+                $params = [
+                    ':publicationStatus' => $publicationStatusId,
+                    ':publicationStatusType' => QubitTerm::STATUS_TYPE_PUBLICATION_ID,
+                    ':id' => $this->resource->id,
+                ];
+
+                $descriptionsUpdated = QubitPdo::modify(
+                    $sql,
+                    $params
+                );
+
+                $response = QubitSearch::getInstance()->updateByQuery(
+                    [
+                        '_id' => $this->resource->id,
+                    ],
+                    [
+                        'script' => [
+                            'inline' => 'ctx._source.publicationStatusId = '.$publicationStatusId,
+                            'lang' => 'painless',
+                        ],
+                    ],
+                    [
+                        // Set to 'proceed' so index update does not abort if a document version
+                        // conflict occurs. This happens if an ES doc is updated during this update.
+                        // Ignore conflicts since conflict documents would still be indexed with the
+                        // updated publication status from the DB.
+                        'conflicts' => 'proceed',
+                    ]
+                );
+
```

Comment:
I think it's better to use a partial update here since it's only a single document update ...

```php
QubitSearch::getInstance()->partialUpdate(
    $this->resource,
    ['publicationStatusId' => $publicationStatusId]
);
```

Reply:
👍